### DISCUSSION TITLE
Web Dom stubs, aka pyi for module js

### DISCUSSION THREAD

**Discussion by simonegiacomelli** [pyodide/pyodide#3470] [Q&A]
Let me start saying that I'm very grateful to all the people that worked and are working on Pyodide :)
It's an amazing project and I'm having a lot of fun using it.

I'm prototyping a library to facilitate the deploy and interoperability between python/server and python/remote (Pyodide).
To feed my IDE auto-completion about the Dom structure, I manually wrote a super simple js.pyi (see below for an example).

Now I'm looking into something more involved like a .pyi generator from webidls like these:
- https://github.com/microsoft/TypeScript-DOM-lib-generator
- https://github.com/rustwasm/wasm-bindgen/tree/main/crates/web-sys

Do you know if an effort to create a 'js.pyi' like the projects above already exists?
So I could join forces instead of building something from scratch.

    class Node:...
    
    class HTMLElement:
        innerHTML: str
        onclick: Any
    
    class Document(Node):
        body: HTMLElement
        def createElement(self, tag: str) -> HTMLElement: ...

**Comment by hoodmane** [comment]
Closed in favor of #3672. Thanks @simonegiacomelli!
